source/_integrations/weather.template.markdown

```diff
@@ -54,6 +55,10 @@ temperature_template:
   description: The current temperature.
   required: true
   type: template
+temperature_unit:
+  description: Unit for temperature.
```

Comment:
We should explain that:
- This is the unit of the template output, not what the user sees
- The output will be converted according to the user's unit system or entity override, we should explain it in https://developers.home-assistant.io/docs/core/entity/weather and link to it
- The valid options are °C, °F and K